### ISSUE TITLE
Make all commands work properly with the second list view

### ISSUE THREAD

**Issue by linhany** [CS2103AUG2016-W13-C4/main#62]
Currently when on view done, delete, edit etc will target the original list, not done list. What's the intended behavior for list done => any other command other than list

**Comment by linhany** [comment]
@CS2103AUG2016-W13-C4/developers

**Comment by linhany** [comment]
For now, any other command will switch list view back to normal. Think that makes sense

**Comment by linhany** [comment]
Maybe we can support a "list done clear" command later to clear the list done

**Comment by edmundmok** [comment]
how about make `clear` target the current view, either on undone or done view then clear accordingly.
`delete` also. Reject all other command input while in `list done` view?

possible `undone` command ?

**Comment by ylhlh** [comment]
Imo that makes things too complicated.
1) Using `clear` to target current view can lead users into clearing the wrong view. Even though we have `undo` such a design flaw should be avoided. E.g. user clears wrong view, didn't realise, quits the app. `undo` will not work now.
2) It makes sense for us to revert back to the original list upon entering any command when in `list done`, due to the fact that we treat the `list done` view as a readonly view. Since we only allow users to `list done clear`,  there is no point in disallowing user from entering commands, forcing them to take an extra step to switch back to done with `undone`. Think in terms of user's perspective. Basically: If both we and the user know what they are trying to do, there is no point in disallowing them to do what they want, making them take an extra step. So we just have to make it clear to the user that the archived list is readonly, since it doesn't make sense for them to edit or add to this list anyway. And the only command we support would be the one that clears this readonly list.

**Comment by linhany** [comment]
1) Using `clear` to target current view can lead users into clearing the wrong view. Even though we have undo such a design flaw should be avoided. E.g. user clears wrong view, didn't realise, quits the app. undo will not work now.
2) Should we support `delete` and `undone` in list done view. I think it is sufficient just to support `list done clear`. Also if we only support `list done clear`, it would simplify things for the user. They won't delete from the wrong list. They won't need to switch from list to list. The list done will be like a readonly thing that won't interfere with their normal operations.

**Comment by edmundmok** [comment]
What if the user thinks that he can delete from the list done view? He enters `delete 3` while in list done view, and enters, then quits the app. Then the 3rd task in the original list is gone without him noticing. Isn't this the same as the problem you mentioned?

I think it would be better to have `clear done` in this case then, `list done clear` seems to be listing a combination of "done" and "clear".

**Comment by linhany** [comment]
We can't remove the possibility of that happening, but we can reduce the chance. Firstly typing `delete 3` switches the list back to normal list upon pressing enter, plus user can see from the message output that he has deleted a task from his normal list, a big hint that something went wrong. Secondly, we can make it clear to the user that the `list done` view is readonly and you cannot edit/delete it, in the user guide under `list done` command. Thirdly, when in list done view we can show in the tooltip "This list done view is not modifiable, entering commands will affect the original list".

Ultimately it comes down to what features we want in the `list done` view. Is it necessary to support individual `delete` and `undone` there?

**Comment by edmundmok** [comment]
The only way the user can tell is through the result display, which similar to the clear command can have a display saying you have cleared the done/undone list appropriately. In fact, if the action is on the current list the effect would be more obvious than if it were on the original list since I cannot remember what was in my original list so I cannot tell if something was missing from it upon switching back (well other than the result display, but assuming it was ignored like in the clear command case). Furthermore, if it was on the current list, it seems more intuitive as my actions are on my current view.

IN conclusion, it should support `delete` on the done list. `undone` is just an idea and might be excessive. Some other commands also have to target the done list anyway, e.g. find, select.

**Comment by linhany** [comment]
Do we still have `select`? We should finalise what commands are supported in the done list. `delete` `find` `clear`? How does `undo` and `redo` work here, is it a separate stack?

**Comment by edmundmok** [comment]
select can probably leave in? its another form of navigation also. yea seems like delete find n clear only. undo and redo can add another variable that indicates action on either undone or done list and take action appropriately.
